src/components/transaction/TransactionForm.js

```diff
@@ -1,260 +1,206 @@
-import React, { Fragment, useEffect, useState, useContext } from 'react';
-import TransactionContext from '../../context/transaction/transactionContext';
-import Papa from 'papaparse';
-
-const TransactionForm = () => {
-  const transactionContext = useContext(TransactionContext);
-
-  const { addTransaction } = transactionContext;
+import React, { Fragment, useEffect, useState } from "react";
+import uuid from "uuid";
+import moment from "moment";
+import Papa from "papaparse";
+
+const TransactionForm = (props) => {
+  const buildTransaction = (transaction) => {
+    // set unique id
+    transaction.id = uuid.v4();
+
+    // format input data
+    transaction.transDate = moment(transaction.transDate);
+    transaction.qty = parseFloat(transaction.qty);
+    transaction.amount = parseFloat(transaction.amount);
+    transaction.fee = parseFloat(transaction.fee);
+
+    // set display date
+    transaction.displayDate = transaction.transDate.format("llll");
+    return transaction;
+  };
 
   useEffect(() => {
-    // // Initializing datePicker and select elements with Materialize-css
```

Comment:
unrelated comment removal. i'm fine if you want to leave it in this PR, but something to keep in mind

Reply:
Noted.